Author: Volker Krause <vkrause@kde.org>
Author: David Faure <faure@kde.org>
Description: Don't keep reading when we reached the end of the stream.
 .
 This should prevent corrupt change replay files from eating up all your
 memory.
Applied-Upstream: 4.9.3, commit:f142cc25b5620c2d1037cc6cd270b1cf301b5380
Bug: https://bugs.kde.org/show_bug.cgi?id=298257

--- a/akonadi/changerecorder_p.h
+++ b/akonadi/changerecorder_p.h
@@ -138,7 +138,7 @@ class AKONADI_TESTS_EXPORT Akonadi::Chan
       QQueue<NotificationMessage> list;
 
       stream >> size;
-      for ( qulonglong i = 0; i < size; ++i ) {
+      for ( qulonglong i = 0; i < size && !stream.atEnd(); ++i ) {
         NotificationMessage msg;
 
         stream >> sessionId;
@@ -186,7 +186,7 @@ class AKONADI_TESTS_EXPORT Akonadi::Chan
         QStringList list;
 
         stream >> size;
-        for ( qulonglong i = 0; i < size; ++i ) {
+        for ( qulonglong i = 0; i < size && !stream.atEnd(); ++i ) {
           stream >> sessionId;
           stream >> type;
           stream >> operation;
